### PR TITLE
Misc work

### PR DESCRIPTION
**Pull request by sdkrystian**

### CODE REVIEW

File: include/mrdox/Metadata/Symbols.hpp [hunk 1]
```diff
@@ -25,7 +25,15 @@ namespace mrdox {
     USR. A USRs is a string that provide an
     unambiguous reference to a symbol.
 */
-using SymbolID = std::array<unsigned char, 20>;
+struct SymbolID :
+    std::array<std::uint8_t, 20>
```
[the remark this answers is not in the document]
Comment (sdkrystian) [reply include/mrdox/Metadata/Symbols.hpp:29]: We don't this is just a (temporary) bandaid until a full rewrite is done

File: include/mrdox/Metadata/Symbols.hpp [hunk 2]
```diff
@@ -14,6 +14,7 @@
 
 #include <mrdox/Platform.hpp>
 #include <llvm/ADT/StringRef.h>
+#include <llvm/ADT/StringExtras.h>
```
Comment (vinniefalco) [inline include/mrdox/Metadata/Symbols.hpp:17]: well don't be adding stuff from llvm you need to be removing it !

File: include/mrdox/Metadata/Symbols.hpp [hunk 3]
```diff
@@ -25,7 +26,65 @@ namespace mrdox {
     USR. A USRs is a string that provide an
     unambiguous reference to a symbol.
 */
-using SymbolID = std::array<unsigned char, 20>;
+class SymbolID
+{
+    // std::uint8_t data_[20];
+    std::array<std::uint8_t, 20> data_;
```
Comment (vinniefalco) [inline include/mrdox/Metadata/Symbols.hpp:32]: fuck, array again? why not `std::uint8_t data_[20];`?

File: include/mrdox/Metadata/Symbols.hpp (hunk 2 above)
Comment (sdkrystian) [reply include/mrdox/Metadata/Symbols.hpp:17]: Oops, forgot to remove that

File: include/mrdox/Metadata/Symbols.hpp (hunk 3 above)
Comment (sdkrystian) [reply include/mrdox/Metadata/Symbols.hpp:32]: @vinniefalco `SHA1::hash` returns a `std::array<std::uint8_t, 20>`. Additionally, we extensively use `SymbolID` as a container (`operator[]`, `begin()`, `end()`, etc).

Comment (sdkrystian) [reply include/mrdox/Metadata/Symbols.hpp:32]: Actually... I may be able to get rid of it... Let's see

File: include/mrdox/Metadata/Symbols.hpp [hunk 4]
```diff
@@ -25,7 +25,80 @@ namespace mrdox {
...
-using SymbolID = std::array<unsigned char, 20>;
+class SymbolID
+{
+    std::uint8_t data_[20];
+
+public:
+    constexpr SymbolID() = default;
+
+    template<typename Elem>
+    SymbolID(const Elem* src)
```
Comment (vinniefalco) [inline include/mrdox/Metadata/Symbols.hpp:36]: I mean this is a little scary but fuck it, this ain't Boost :)

Comment (sdkrystian) [reply include/mrdox/Metadata/Symbols.hpp:36]: LOL yeah

**Comment by sdkrystian** [comment]
@vinniefalco everything look good here?

**Comment by vinniefalco** [comment]
If I said no what would change? lol

**Comment by sdkrystian** [comment]
Are there any changes you want?

**Pull request merged**